lib/Gedmo/Sortable/Mapping/Event/Adapter/ORM.php

```diff
@@ -36,15 +37,54 @@ private function addGroupWhere(QueryBuilder $qb, $groups)
         $i = 1;
         foreach ($groups as $group => $value) {
             if (null === $value) {
-                $qb->andWhere($qb->expr()->isNull('n.'.$group));
+                $qb->andWhere($qb->expr()->isNull('n.' . $group));
             } else {
-                $qb->andWhere('n.'.$group.' = :group__'.$i);
-                $qb->setParameter('group__'.$i, $value);
+                $qb->andWhere('n.' . $group . ' = :group__' . $i);
+                $qb->setParameter('group__' . $i, $this->getGroupValue($value), $this->getGroupType($value));
             }
             $i++;
         }
     }
 
+    /**
+     * @param $value
+     * @return bool
+     */
+    private function isEntity($value)
+    {
+        return (is_object($value) && $this->getObjectManager()->getMetadataFactory()->hasMetadataFor(ClassUtils::getClass($value)));
+    }
+
+    /**
+     * @param $value
+     * @return mixed
+     */
+    private function getGroupValue($value)
+    {
+        if (!$this->isEntity($value)) {
+            return $value;
+        }
+        return $this->getObjectManager()->getUnitOfWork()->getSingleIdentifierValue($value);
+    }
+
+    /**
+     * @param $value
+     * @return \Doctrine\DBAL\Types\Type|null|string
+     */
+    private function getGroupType($value)
+    {
+        if (!$this->isEntity($value)) {
+            return null;
```

Comment:
Please return $metaData->getTypeOfField($group) here ;)

Like I said, custom type value objects are also a possibility. Infer type will not be able to resolve those, and the sortable won't work.